cpp/bench/ann/CMakeLists.txt

```diff
@@ -15,9 +15,18 @@
 # ##################################################################################################
 # * benchmark options ------------------------------------------------------------------------------
 
-option(RAFT_ANN_BENCH_USE_FAISS_BFKNN "Include faiss' brute-force knn algorithm in benchmark" ON)
+option(RAFT_ANN_BENCH_USE_FAISS_FLAT "Include faiss' brute-force knn algorithm in benchmark" ON)
 option(RAFT_ANN_BENCH_USE_FAISS_IVF_FLAT "Include faiss' ivf flat algorithm in benchmark" ON)
 option(RAFT_ANN_BENCH_USE_FAISS_IVF_PQ "Include faiss' ivf pq algorithm in benchmark" ON)
+option(RAFT_ANN_BENCH_USE_FAISS_CPU_FLAT
+       "Include faiss' cpu brute-force knn algorithm in benchmark" ON
+)
+option(RAFT_ANN_BENCH_USE_FAISS_CPU_FLAT "Include faiss' cpu brute-force algorithm in benchmark" ON)
+
+option(RAFT_ANN_BENCH_USE_FAISS_CPU_IVF_FLAT "Include faiss' cpu ivf flat algorithm in benchmark"
+       ON
+)
+option(RAFT_ANN_BENCH_USE_FAISS_CPU_IVF_PQ "Include faiss' cpu ivf pq algorithm in benchmark" ON)
```

Comment:
Since this is a rather big, breaking change by now, would it make sense to standardize the FAISS algo names?:
```
CMake OPTION                             executable name                algo name
RAFT_ANN_BENCH_USE_FAISS_GPU_FLAT        FAISS_GPU_FLAT_ANN_BENCH       faiss_gpu_flat
RAFT_ANN_BENCH_USE_FAISS_GPU_IVF_FLAT    FAISS_GPU_IVF_FLAT_ANN_BENCH   faiss_gpu_ivf_flat
RAFT_ANN_BENCH_USE_FAISS_GPU_IVF_PQ      FAISS_GPU_IVF_PQ_ANN_BENCH     faiss_gpu_ivf_pq
RAFT_ANN_BENCH_USE_FAISS_CPU_FLAT        FAISS_CPU_FLAT_ANN_BENCH       faiss_cpu_flat
RAFT_ANN_BENCH_USE_FAISS_CPU_IVF_FLAT    FAISS_CPU_IVF_FLAT_ANN_BENCH   faiss_cpu_ivf_flat
RAFT_ANN_BENCH_USE_FAISS_CPU_IVF_PQ      FAISS_CPU_IVF_PQ_ANN_BENCH     faiss_cpu_ivf_pq
```
i.e. consistently add the CPU/GPU prefixes to both versions. This would make it easier to do the algorithm name checks (e.g. executable name -> to lower case).
(At this point, it seems the cmake options specify only "CPU", but algo names in configs specify only "gpu")

Reply:
+1 for explicit CPU/GPU prefix. BTW, what is the motivation to have so many separate executables?

Reply:
Yes, I plan to fix the naming so that it's more consistent. 

> what is the motivation to have so many separate executables?

The reason we did this is because we had some conflicting compile flags across binaries- specifically between things like HNSW and FAISS. I think it's better to allow each library to have their own individual binaries. I think we should also consolidate the binaries within each library but we're butting up against the release here and I'd like to do that consolidation in 23.12 since it's tech debt and not impacting the operation of the benchmarks.